### ISSUE TITLE
[pos_expenses_pay] Things to do

### ISSUE THREAD

**Issue by janbec** [it-projects-llc/addons-dev#388]
My feedback @ilmir-k 

## 1. Sale Details Report
![image](https://user-images.githubusercontent.com/28890800/37476036-669a5596-2874-11e8-9362-6f1fbbef1e54.png)

- Please add a row for the sum of the expenses
- please change term partner to employee

## 2. Expenses tree view in PoS
- display payment_mode in column

## 3. Pay Expenses message
- the header should reflect payment_mode (Reimburse Employee or Company)

## 4. Automatic Email
- Could not find anything indicating the function has been implemented?

## 5. Account move lines
- CE, could not check anything but don't think anything was changed regarding this?

**Comment by ilmir-k** [comment]
Hi @janbec 

> 5. Account move lines

Right, we didn't change anything for account move lines. It has built-in behavior.

====
I have some questions:

> 3. Pay Expenses message

Is that about the popup message in POS ?

> 4. Automatic Email

Please could you clear this point? What automatic email do you mean?

**Comment by janbec** [comment]
## 3. Pay Expenses Message
![image](https://user-images.githubusercontent.com/28890800/37722030-57c6c098-2d2b-11e8-87ad-cfbf6399af74.png)

## 4. Automatic Email
If an employee uses the expenses mail and sends in an attachment (receipt), I would like to automatically forward that attachment to our electronic archive.

This can happen when the manager approves or the expense is paid (plan B).

The email address is fix. If it speeds up the process the mails can be send from the queue.

**Comment by ilmir-k** [comment]
@janbec yes, I received it without mention too.
I still don't understand the point "4. Automatic Email"

I know that the user can attach any attachment (receipt) manually:
![attach](https://user-images.githubusercontent.com/14342645/37760411-63be2744-2dd8-11e8-85df-8bdaca4bf800.png)

And we have expense messages with attachment:
![expense_messages](https://user-images.githubusercontent.com/14342645/37761196-d070b4b8-2dda-11e8-8619-c74a11ea569e.png)

We should make any actions with attachments in expense messages?
What do you mean under "our electronic archive"?

Please give us more clarification. Thanks

**Comment by janbec** [comment]
Sorry, probably should have given more context.

In expenses settings, you can configure an alias for expenses. If the employee sends an mail with his receipt as a picture attachted to that email address (e.g. expenses@company.com), odoo will create an expenses and attach the picture of the receipt.

If the expense has a receipt, I would like odoo to send an email (expenses@dms.com) to our document management system which is able to archive the document then.

Is that more understandable?

**Comment by ilmir-k** [comment]
@janbec I understand now. Thanks for the clarification.
We will analyze this question.

**Comment by ilmir-k** [comment]
Hi @janbec 

You can test the latest updates:
http://runbot.it-projects.info/runbot/33/pos-addons-11.0-pos_expense_pay
login: admin
pass: admin
It contains all the updates except Automatic Email.

Regarding the automatic email feature, what do you think if we would use built-in odoo channels for this.
For example, once the expense is created, email will be sent to the corresponding channel and to followers of this channel:

![channel1](https://user-images.githubusercontent.com/14342645/38030745-e7333068-32b2-11e8-86c8-687b9a76204b.png)

![channel2](https://user-images.githubusercontent.com/14342645/38030747-e77fdf80-32b2-11e8-89d2-563af94df1e7.png)

![channel3](https://user-images.githubusercontent.com/14342645/38030748-e7da182e-32b2-11e8-8199-798d838c04d9.png)

![expense1](https://user-images.githubusercontent.com/14342645/38030766-f3e933de-32b2-11e8-856e-d72dcb4c4027.png)

![expense2](https://user-images.githubusercontent.com/14342645/38030769-f48c9b64-32b2-11e8-9e6d-e1e65ccaccef.png)


In this case any attachment from expense mail thread will be shown on the channel.
We need just to think how to display attachment on the channel which is not in mail thread

![expense3](https://user-images.githubusercontent.com/14342645/38031504-bdc7eae6-32b4-11e8-8a4a-6a2bb672849e.png)

![channel4](https://user-images.githubusercontent.com/14342645/38031522-c41c5288-32b4-11e8-922d-2c465f1cdd11.png)

**Comment by janbec** [comment]
Hi Ilmir,
everything looks good!

Using the channels is a great idea! I would suggest to either:

1. Create a new view based on mail.view_document_file_kanban for the documents button in expenses with nocreate
2. Plan B would be to invisible the button to prevent creation on that way

Let me know what you think.

**Comment by ilmir-k** [comment]
Hi @janbec 
I think it's enough to invisible the Document button to prevent such creation.

**Comment by ilmir-k** [comment]
@janbec 
Could you test the latest updates? 
http://runbot.it-projects.info/runbot/33/pos-addons-11.0-pos_expense_pay
Should we make something else?

**Comment by janbec** [comment]
Hi Ilmir,
Hi Dinar,

I see that you used xml to invisible all means to attach a file without it going to the channel set as follower with 1fbe4b04d692bc1182615a5500baf412a74c2da6

However, I noticed that if you only create an internal note (not message) the attachment/message is not shown in the channel. I realize that this is expected behavior...

Further, I was not able to produce a failing outgoing mail to a follower I had set in a channel (Technical -> Email). Do you have test mail servers to configure outgoing and incoming servers in runbot?

I'm worried that when I submit by mail the expense is created completely before a follower is added...

**Comment by ilmir-k** [comment]
Hi Jan,

> Further, I was not able to produce a failing outgoing mail to a follower I had set in a channel (Technical -> Email). Do you have test mail servers to configure outgoing and incoming servers in runbot?

We have not such server now. We'll try to setup it.

**Comment by ilmir-k** [comment]
@janbec 
It would be great if you could test it in your environment, because it requires a time and some tricks to setup server on runbot. Thanks.

**Comment by ilmir-k** [comment]
@janbec 
Have you been able to test the module on your environment?

**Comment by janbec** [comment]
Not really. However, I staged the situation with the recruitment module and email alias for that. I couldn't get it to work as desired. Adding a channel automatically is not possible via automatic action (except for execute python code via server action) and a reply with document to that mail thread was not forwarded to the channel

But I noticed I could just create a res.partner and add him automatically (easily done with automatic action). He will get the first mail that creates a document with attachment for sure.

I think that should solve the problem easily....
